### PR TITLE
[LPS-119498, LPS-119495, LPS-119808] Modal Sizes and Navigation Menu fix for adding new items

### PR DESCRIPTION
**Pull request by jbalsas**
Hey @liferay-echo, here's a bit of a mix of different issues. I grouped them together to try and save on CI, but they can all be sent separately.

#### [LPS-119498 Sign in Modal is oversized](https://issues.liferay.com/browse/LPS-119498)

The title says it all... I'm sending this one over because the code triggering it is in some of our modules, but maybe we want to find a way to put it elsewhere in the future...

Screen Shot 2020-08-27 at 08 40 13

#### [LPS-119499 Add Page Modal is too big](https://issues.liferay.com/browse/LPS-119499)

This one also explains itself... when the new modals were used as part of [Allow the user to create a new collection page
](https://issues.liferay.com/browse/LPS-114877) the fixed sizes were removed (as in other places) and at some point we started enforcing that the API when taking a `url` but no `size` should default to `fullscreen` since it's the most common case, probably breaking this. Anyways...

Screen Shot 2020-08-27 at 08 45 29

#### [LPS-119808 Adding an item to an empty NavigationMenu from the New dropdown navigates instead of opening a dialog](https://issues.liferay.com/browse/LPS-119808)

Testing the fix for [LPS-119499 Add Page Modal is too big](https://issues.liferay.com/browse/LPS-119499) I saw this and filed the ticket myself. 

**Steps to reproduce:**
- Navigate to `Site Builder > Navigation Menus`
- Create a new Navigation Menu
- Try to use the `New` dropdown at the bottom to create a new menu item

I feel there were a lot of little hacks that were doing this work before... so I applied the new `propsTransformer` pattern that allows us to pass in `js` behaviour to the new React-based clay components so that we can more easily share specific `onClick` behaviours rather than having to register global instances from custom components as we were doing here.

I think will do a bigger round of refactoring based on this `propsTransformer` concept right after the release, but for now you can see it in action here and also in other parts of DXP that already benefit from it.

### CODE REVIEW

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by jbalsas** [comment]
> ❌ ci:test:sf

🙈

**Comment by jbalsas** [comment]
ci:stop

File: modules/apps/site-navigation/site-navigation-admin-web/src/main/java/com/liferay/site/navigation/admin/web/internal/display/context/SiteNavigationAdminDisplayContext.java [hunk 1]
```diff
@@ -98,9 +98,9 @@ public SiteNavigationAdminDisplayContext(
 					add(
 						dropdownItem -> {
 							dropdownItem.setData(
-								Collections.singletonMap("type", "add-button"));
-							dropdownItem.setHref(
-								_getAddURL(siteNavigationMenuItemType));
+								Collections.singletonMap(
```
Comment (jbalsas) [inline modules/apps/site-navigation/site-navigation-admin-web/src/main/java/com/liferay/site/navigation/admin/web/internal/display/context/SiteNavigationAdminDisplayContext.java:101]: I couldn't find references to this `data-type`... but maybe it's worth keeping?

**Comment by jbalsas** [comment]
ci:test:sf

**Comment by jbalsas** [comment]
ci:test:relevant

**Comment by victorg1991** [comment]
Taking a look :)

File: modules/apps/site-navigation/site-navigation-admin-web/src/main/java/com/liferay/site/navigation/admin/web/internal/display/context/SiteNavigationAdminDisplayContext.java (hunk 1 above)
Comment (victorg1991) [reply modules/apps/site-navigation/site-navigation-admin-web/src/main/java/com/liferay/site/navigation/admin/web/internal/display/context/SiteNavigationAdminDisplayContext.java:101]: Yeah, removing that seems harmless

**Comment by victorg1991** [comment]
LGTM 🎉  Pretty modals again 😍

I've seen a minor styles bug:
Screenshot 2020-08-27 at 11 57 38

I am going to send the fix as a separate PR to avoid the risk of the test relevant failing in next tries

Thanks a lot!

**Comment by victorg1991** [comment]
ci:forward

**Pull request closed without merging**